ComponentKit/Core/CKComponentController.mm

```diff
@@ -66,17 +68,37 @@ static void eraseAnimation(CKAppliedComponentAnimationMap &map, CKComponentAnima
   }
 }
 
+struct CKComponentControllerAnimationData {
+  CKComponentAnimationID nextAnimationID;
+  std::vector<CKPendingComponentAnimation> pendingAnimationsOnInitialMount;
+  CKAppliedComponentAnimationMap appliedAnimationsOnInitialMount;
+  std::vector<CKPendingComponentAnimation> pendingAnimations;
+  CKAppliedComponentAnimationMap appliedAnimations;
+};
+
+struct CKComponentControllerAnimationWrapper {
+public:
+  CKComponentControllerAnimationData *operator->() {
+    if (_animationData == nullptr) {
+      _animationData.reset(new CKComponentControllerAnimationData());
+    }
+    return _animationData.get();
+  };
+
+  explicit operator bool() const {
+    return _animationData != nullptr;
+  };
+private:
+  std::unique_ptr<CKComponentControllerAnimationData> _animationData;
+};
+
 @implementation CKComponentController
 {
   CKComponentControllerState _state;
   BOOL _updatingComponent;
   BOOL _performedInitialMount;
   CKComponent *_previousComponent;
-  CKComponentAnimationID _nextAnimationID;
-  std::vector<CKPendingComponentAnimation> _pendingAnimationsOnInitialMount;
-  CKAppliedComponentAnimationMap _appliedAnimationsOnInitialMount;
-  std::vector<CKPendingComponentAnimation> _pendingAnimations;
-  CKAppliedComponentAnimationMap _appliedAnimations;
+  CKComponentControllerAnimationWrapper _animationData;
```

Comment:
This makes the state managed by the `CKComponentController` much clearer (no need to animation-related state to add clutter). Nice side effect of this work.